pkg/api/validation/validation.go

```diff
@@ -1237,6 +1256,21 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, validateScaleIOVolumeSource(pv.Spec.ScaleIO, specPath.Child("scaleIO"))...)
 		}
 	}
+	if pv.Spec.LocalVolume != nil {
+		if numVolumes > 0 {
+			allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "may not specify more than 1 volume type"))
+		} else {
+			numVolumes++
+			if !utilfeature.DefaultFeatureGate.Enabled(features.PersistentLocalVolumes) {
+				allErrs = append(allErrs, field.Forbidden(specPath.Child("localVolume"), "LocalVolumes are disabled by feature-gate"))
+			}
+			allErrs = append(allErrs, validateLocalVolumeSource(pv.Spec.LocalVolume, specPath.Child("localVolume"))...)
+			// StorageClass is required
+			if len(pv.Spec.StorageClassName) == 0 {
```

Comment:
The main concern I have is that PVC Is a user construct, and the user shouldn't care about the underlying storage topology.  We could push towards a model where StorageClass is required in order to express more advanced storage concepts.